packages/drivers/odsp-driver/src/compactSnapshotParser.ts

```diff
@@ -104,7 +117,9 @@ function readTreeSection(node: NodeCore) {
                 // "name": <node name>
                 // "children": <blob id>
                 if (content === "children") {
-                    trees[treeNode.getString(1)] = readTreeSection(treeNode.getNode(3));
+                    const result = readTreeSection(treeNode.getNode(3));
```

Comment:
Looks good, but I'd run a validation perf run with these changes to see how much impact it has on end-to-end performance. I've noticed that even such small changes could cause quite sizable regression in perf, so knowing the cost of telemetry like that would be great.

Reply:
Here is one idea how to avoid it (though it changes slightly dynamics): we can split readTreeSection() into readTreeSectionFast() & readTreeSectionSlow().
readTreeSectionFast() wouuld assume only fast code path, and when it detects it can't do it at any level it will throw an exception. And on exception, we will restart process with readTreeSectionSlow().

It would only work if we have zero-tolerance policy against slow path. I.e., any single hit we find we investigate and figure out how to keep it on fast path. It would not work if 1% of file opens hit slow path.
That's basically what I do with string parsing. I'd go that route only if your measurement shows it's becoming slower.
But it maybe also worth prototyping it and seeing if we are getting faster overall (even without extra telemetry) - it could be! :)

Reply:
I checked. With current implementation, there is no regression in timing. However, I will try what you suggested in other change. So, merging this.